linera-witty-macros/src/util/mod.rs

```diff
@@ -71,8 +39,19 @@ impl Parse for AttributeParameters {
 }
 
 impl AttributeParameters {
+    /// Parses the attribute parameters to the attribute procedural macro.
+    pub fn new(attribute_parameters: proc_macro::TokenStream) -> Self {
+        syn::parse(attribute_parameters.clone()).unwrap_or_else(|_| {
+            abort!(
+                TokenStream::from(attribute_parameters),
+                r#"Failed to parse attribute parameters, expected either `root = true` \
+                or `package = "namespace:package""#
```

Comment:
Have we considered using [`darling`](https://docs.rs/darling/latest/darling/)?  I think it should give good error messages by default — without us having to manually duplicate the syntax definition between parser/error.

Reply:
Interesting, I didn't know about that crate. I took an initial look, and I think I need to experiment with it a bit first to see what the outcome looks like. I'd rather do this some time later :sweat_smile:

Reply:
If it's a pain — maybe we should make an issue (maybe it's a `good-first-issue`?) and leave it for now :)

Reply:
I've created this issue (https://github.com/linera-io/linera-protocol/issues/1762) and approved :)